plugins/repository-gcs/src/main/java/org/opensearch/repositories/gcs/ProxySettings.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.gcs;
+
+import org.opensearch.common.Strings;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.Objects;
+
+public class ProxySettings {
```

Comment:
Minor - Most of this is identical to the [Azure ProxySettings class](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/ProxySettings.java). Could you move the majority of this to a superclass and declare Azure and GCS specific subclasses?